### PR TITLE
Fix provisional name search issues

### PR DESCRIPTION
**Pull request by mo-nathan**
There were a set of examples provided by Huafeng for searches some of which did not work with the pattern search.  Huafeng's cases:

Amanita sp. ‘amerirubescens04’ Tulloss & K.W. Hughes crypt. temp.
Cortinarius "sp-IN34"
Cortinarius sp. ‘IN34’ S.D. Russell crypt. Temp.
Cortinarius subpulchrifolius
Inocybe "sp-IN125"
Mycena sp. ‘maculata-IN01’
Russula "laurocerasi-IN01"
Russula "sp-IN41"
Russula sp. ‘MO04’
Russula sp. ‘MO04’ S.D. Russell crypt. temp.

This fix pushes name-like strings for Observation and Name search through the Name.parse_name system to standardize provisional names.  This works well for all the above cases.  However, it would not surprise me if this has created some new unintentional issues so it's worth testing this reasonably throughly.  I would like to get this fix out soon for Huafeng.

Note that there are probably other search cases (e.g., Advanced Search) and probably some other cases that may not handle old school provisional names correctly.  I believe a similar approach should work.

### CODE REVIEW

File: app/classes/pattern_search/base.rb [hunk 1]
```diff
@@ -9,7 +9,13 @@ def initialize(string)
       self.errors = []
       self.parser = PatternSearch::Parser.new(string)
       build_query
+      real_model = model.name.to_sym
       self.query = Query.lookup(model.name.to_sym, args)
```
Comment (JoeCohen) [inline app/classes/pattern_search/base.rb:13]: Is line 13  needed (`self.query = Query.lookup(model.name.to_sym, args`)
in light of new line 18 (`self.query = Query.lookup(real_model, args`) ?

**Review by JoeCohen — COMMENTED** [review]
1. I didn't see any issues while playing around with this locally.
2. There are 2 test failures
3. Not part of this PR, but noting it here FYI. Almost certainly an edge case not worth messing with.

This old name: 
Retiboletus clade III sensu Zeng et al. http://localhost:3000/names/10769
had author: "III sensu Zeng et al."
```ruby
#<Name:0x0000000136a78ac8
 id: 107696,
...
 rank: "Group",
 text_name: "Retiboletus clade",
 search_name: "Retiboletus clade III sensu Zeng et al.",
 display_name: "**__Retiboletus__** clade III sensu Zeng et al.",
 sort_name: "Retiboletus   clade  III sensu Zeng et al.",
...
 ok_for_export: true,
 author: "III sensu Zeng et al.",
...>
```

I fed `Retiboletus clade III sensu Zeng et al.` to the parser by 
manually updating the name with 
`Retiboletus clade III sensu Zeng et al.` in the Scientific Name field, and 
the Author field blank.
Result:
```ruby
#<Name:0x0000000136a794c8
 id: 107696,
 ...
 rank: "Group",
 text_name: "Retiboletus sp. 'III' clade",
 search_name: "Retiboletus sp. 'III' clade sensu Zeng et al.",
 display_name: "**__Retiboletus__** sp. **__'III'__** clade sensu ...",
 sort_name: "Retiboletus iii   clade  sensu Zeng et al.",
 ...
 author: "sensu Zeng et al.",
... nil>
```

That's a big improvement (`author` is correct).
The `text_name` is not wrong, but seems weird.
In a perfect world I think the text_name would be "Retiboletus Clade III" (the cited article has "Clade III") or "Retiboletus clade III".
(I wonder what Steve Russell or Conrad Schoch would like. Maybe this like what the MO parser produces.)
Probably not worth messing with.
Screenshot 2025-06-24 at 8 32 16 AM

File: app/classes/pattern_search/base.rb (hunk 1 above)
Comment (mo-nathan) [reply app/classes/pattern_search/base.rb:13]: Good catch.  Fat fingered that one.

**Comment by mo-nathan** [comment]
Regarding `Retiboletus clade III sensu Zeng et al.`, You might try `Retiboletus subg. 'clade-III'`.  I'm definitely not going to implement something new to support that any time soon.

File: app/classes/pattern_search/observation.rb [hunk 2]
```diff
@@ -90,8 +90,8 @@ def default_to_including_synonyms_and_subtaxa
     end
 
     def is_pattern_a_name?
-      ::Name.where("text_name = ? OR search_name = ?",
-                   args[:pattern].to_s, args[:pattern].to_s).any?
+      pat = ::Name.parse_name(args[:pattern])&.search_name || args[:pattern]
+      ::Name.where("text_name = ? OR search_name = ?", pat, pat).any?
```
[the remark this answers is not in the document]
Comment (mo-nathan) [reply app/classes/pattern_search/observation.rb]: Sure.  Done.  I was just trying to get something working with minimal changes.

**Review by JoeCohen — APPROVED** [review]
